types/rational.go

```diff
@@ -70,7 +73,17 @@ func NewRatFromDecimal(decimalStr string) (f Rat, err Error) {
 
 	num, errConv := strconv.Atoi(numStr)
 	if errConv != nil {
-		return f, ErrUnknownRequest(errConv.Error())
+		// resort to big int, don't make this default option for efficiency
```

Comment:
I feel like a better thing to do would be to check the specific type of error - only if it was an (overflow?) error then we would go to this next step... shouldn't have to go inside here if the user is attempting to convert "alskgndnalsg" into a rational